File: megumi/src/main/java/com/erisu/cloud/megumi/command/MethodLite.java
package com.erisu.cloud.megumi.command;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import java.lang.reflect.Method;

/**
 * @Description 用以方法反射, bean为注入至ioc中bean的节点
 * @Author alice
 * @Date 2021/6/3 19:42
 **/
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MethodLite {
    private Method method;
    private Object bean;
}

File: megumi/src/main/java/com/erisu/cloud/megumi/command/MethodLite.kt
package com.erisu.cloud.megumi.command

import net.mamoe.mirai.message.data.Message
import java.lang.reflect.Method
import kotlin.reflect.KCallable
import kotlin.reflect.KFunction
import kotlin.reflect.full.callSuspend
import kotlin.reflect.full.callSuspendBy

/**
 * @Description 用以方法反射, bean为注入至ioc中bean的节点
 * @Author alice
 * @Date 2021/6/3 19:42
 **/
data class MethodLite(
    val method: Method,
    val bean: Any,
)


